### ISSUE TITLE
Ignore chef.io backend problems

### ISSUE THREAD

**Issue by zepheiryan** [chef-cookbooks/chef_client_updater#82]
### Cookbook version
3.1.2

### Chef-client version
12.21.26

### Platform Details
Ubuntu 16.04.3

### Scenario:
When the chef.io infrastructure experiences problems as it has a few times in the past week, the `chef_client_updater` resource fails the entire Chef run due to an inability to connect to the backend.  It would be useful to have an option to ignore this specific type of connection failure as one hopes it's just a transient problem and isn't something directly fixable by Chef users anyways.

I was hoping there was a way to use `edit_resource` with `ignore_failure` to do what I was looking for here, but the exception seems to be from Ruby code, not a resource (though I would be happy to be told otherwise).

### Steps to Reproduce:
Include this cookbook's default recipe in a run list and... well, wait for packages.chef.io to go down, then do a Chef run.

### Expected Result:
Not so much expectation as a request - log a connection warning but don't treat it as a failing condition.

### Actual Result:

Run fails with this error
```
[2018-01-16T00:50:26+00:00] INFO: Processing chef_client_updater[update chef-client] action update (chef_client_updater::default line 20)
[2018-01-16T00:50:27+00:00] WARN: NO /opt/chef.upgrade DIR TO ROLL BACK TO!
...
[2018-01-16T00:50:28+00:00] ERROR: chef_client_updater[update chef-client] (chef_client_updater::default line 20) had an error: Net::HTTPFatalError: 503 "Backend is unhealthy"
[2018-01-16T00:50:28+00:00] ERROR: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
```

and top lines of the stacktrace:
```
>>>> Caused by Net::HTTPFatalError: 503 "Backend is unhealthy"
/opt/chef/embedded/lib/ruby/2.3.0/net/http/response.rb:120:in `error!'
/opt/chef/embedded/lib/ruby/2.3.0/net/http/response.rb:129:in `value'
/opt/chef/embedded/lib/ruby/gems/2.3.0/gems/mixlib-install-3.8.0/lib/mixlib/install/backend/package_router.rb:156:in `get'
/opt/chef/embedded/lib/ruby/gems/2.3.0/gems/mixlib-install-3.8.0/lib/mixlib/install/backend/package_router.rb:125:in `artifacts_for_version'
/opt/chef/embedded/lib/ruby/gems/2.3.0/gems/mixlib-install-3.8.0/lib/mixlib/install/backend/package_router.rb:43:in `available_artifacts'
/opt/chef/embedded/lib/ruby/gems/2.3.0/gems/mixlib-install-3.8.0/lib/mixlib/install/backend/base.rb:62:in `info'
/opt/chef/embedded/lib/ruby/gems/2.3.0/gems/mixlib-install-3.8.0/lib/mixlib/install/backend.rb:29:in `info'
/opt/chef/embedded/lib/ruby/gems/2.3.0/gems/mixlib-install-3.8.0/lib/mixlib/install.rb:46:in `artifact_info'
/var/chef/cache/cookbooks/chef-ingredient/libraries/default_handler.rb:99:in `configure_from_channel'
/var/chef/cache/cookbooks/chef-ingredient/libraries/default_handler.rb:49:in `configure_package'
/var/chef/cache/cookbooks/chef-ingredient/libraries/default_handler.rb:23:in `handle_install'
/var/chef/cache/cookbooks/chef-ingredient/resources/chef_ingredient.rb:59:in `block in class_from_file'
```

**Comment by martinisoft** [comment]
I'd like to suggest an additional solution. If you specify a specific version, check if it is installed and short-circuit at that point which will save the outbound call.

**Comment by onlyhavecans** [comment]
We have also been having this issue and it is breaking a lot of runs. I like @martinisoft 's solution

**Comment by zepheiryan** [comment]
Maintainers implemented @martinisoft 's suggestion in #83 and released a couple days ago.  AIUI, it would still throw an error if the exact desired version doesn't match and the Chef backend is down - which sounds like the right behavior to me.  Closing.
